compiler/rustc_trait_selection/src/error_reporting/infer/mod.rs

```diff
@@ -1693,7 +1804,7 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
 
         // It reads better to have the error origin as the final
         // thing.
-        self.note_error_origin(diag, cause, exp_found, terr);
+        self.note_error_origin(diag, cause, exp_found, terr, param_env);
```

Comment:
Instead of passing in `param_env` in all of the callers here, you can extract the param-env from the `values` argument that already has one with `values.map(|values| values.param_env)`.